### PR TITLE
refactor: remove internal TemplateRenderer use from VirtualList

### PR DESCRIPTION
**Pull request by tomivirkki**
## Description

Replace `TemplateRenderer` with `LitRenderer` in `VirtualList` 

Part of https://github.com/vaadin/flow-components/issues/4059

## Type of change

- Refactoring

### CODE REVIEW

**Comment by sissbruecker** [comment]
> Closes #4059

`TemplateRenderer` is still used in `TreeGrid`:
https://github.com/vaadin/flow-components/blob/2ea83688c3b460580ac09c74ad013c2f50784a17/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java#L558-L565

Also we might want to remove it's usage from ITs, and the public API itself.

**Comment by tomivirkki** [comment]
> TemplateRenderer is still used in TreeGrid

Ah yes, indeed. This will be removed separately

File: vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListIT.java [hunk 1]
```diff
@@ -459,8 +463,9 @@ private void assertItemsArePresent(JsonArray items, int startingIndex,
                     "Object at index " + i + " is null, when it shouldn't be",
                     items.get(i),
                     CoreMatchers.not(CoreMatchers.instanceOf(JsonNull.class)));
+            System.out.println(items.getObject(i));
```
Comment (sissbruecker) [inline vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListIT.java:466]: Let's remove the debug logging here.

**Review by sissbruecker — APPROVED** [review]

File: vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListIT.java (hunk 1 above)
Comment (tomivirkki) [reply vaadin-virtual-list-flow-parent/vaadin-virtual-list-flow-integration-tests/src/test/java/com/vaadin/flow/component/virtuallist/tests/VirtualListIT.java]: Done

**Pull request merged**